### PR TITLE
Add a demo page to auth-exp using the new modular methods

### PR DESCRIPTION
**Pull request by samhorlbeck**
Note: I just took the old demo (in `packages/auth/demo`) and modified it a bit. It has a ton of code and I don't think it's worth our time to try to modernize all of that at the moment.

### CODE REVIEW

File: packages-exp/auth-exp/demo/src/index.js [hunk 1]
```diff
@@ -0,0 +1,1905 @@
...
+  } catch (error) {
+    alertError('Error: ' + error.code);
+  }
+}
+
+/**
+ * Signs up a new user with an email and a password.
+ */
+function onSignUp() {
+  alertNotImplemented();
```
Comment (avolkovi) [inline packages-exp/auth-exp/demo/src/index.js:315]: this is implemented now

File: packages-exp/auth-exp/demo/src/index.js [hunk 2]
```diff
@@ -0,0 +1,1905 @@
...
+  // var password = $('#signup-password').val();
+  // auth.createUserWithEmailAndPassword(email, password)
+  //     .then(onAuthUserCredentialSuccess, onAuthError);
+}
+
+/**
+ * Signs in a user with an email and a password.
+ */
+function onSignInWithEmailAndPassword() {
+  alertNotImplemented();
```
Comment (avolkovi) [inline packages-exp/auth-exp/demo/src/index.js:326]: this is implemented now

File: packages-exp/auth-exp/demo/src/index.js [hunk 3]
```diff
@@ -0,0 +1,1905 @@
...
+    refreshUserData();
+    alertSuccess('Profile updated!');
+  }, onAuthError);
+}
+
+/**
+ * Sends sign in with email link to the user.
+ */
+function onSendSignInLinkToEmail() {
+  alertNotImplemented();
```
Comment (avolkovi) [inline packages-exp/auth-exp/demo/src/index.js:718]: this is implemented now

File: packages-exp/auth-exp/demo/src/index.js [hunk 4]
```diff
@@ -0,0 +1,1905 @@
...
+  // auth.sendSignInLinkToEmail(email, getActionCodeSettings()).then(function() {
+  //   alertSuccess('Email sent!');
+  // }, onAuthError);
+}
+
+/**
+ * Sends sign in with email link to the user and pass in current url.
+ */
+function onSendSignInLinkToEmailCurrentUrl() {
+  alertNotImplemented();
```
Comment (avolkovi) [inline packages-exp/auth-exp/demo/src/index.js:729]: this is implemented now

File: packages-exp/auth-exp/demo/src/index.js [hunk 5]
```diff
@@ -0,0 +1,1905 @@
...
+  //   }
+  //   alertSuccess('Email sent!');
+  // }, onAuthError);
+}
+
+/**
+ * Sends email link to link the user.
+ */
+function onSendLinkEmailLink() {
+  alertNotImplemented();
```
Comment (avolkovi) [inline packages-exp/auth-exp/demo/src/index.js:754]: this is implemented now

File: packages-exp/auth-exp/demo/src/index.js [hunk 6]
```diff
@@ -0,0 +1,1905 @@
...
+  // auth.sendSignInLinkToEmail(email, getActionCodeSettings()).then(function() {
+  //   alertSuccess('Email sent!');
+  // }, onAuthError);
+}
+
+/**
+ * Sends password reset email to the user.
+ */
+function onSendPasswordResetEmail() {
+  alertNotImplemented();
```
Comment (avolkovi) [inline packages-exp/auth-exp/demo/src/index.js:765]: this is implemented now

File: packages-exp/auth-exp/demo/src/index.js [hunk 7]
```diff
@@ -0,0 +1,1905 @@
...
+  // auth.sendPasswordResetEmail(email, getActionCodeSettings()).then(function() {
+  //   alertSuccess('Email sent!');
+  // }, onAuthError);
+}
+
+/**
+ * Verifies the password reset code entered by the user.
+ */
+function onVerifyPasswordResetCode() {
+  alertNotImplemented();
```
Comment (avolkovi) [inline packages-exp/auth-exp/demo/src/index.js:776]: this is implemented now

File: packages-exp/auth-exp/demo/src/index.js [hunk 8]
```diff
@@ -0,0 +1,1905 @@
...
+  // auth.verifyPasswordResetCode(code).then(function() {
+  //   alertSuccess('Password reset code is valid!');
+  // }, onAuthError);
+}
+
+/**
+ * Confirms the password reset with the code and password supplied by the user.
+ */
+function onConfirmPasswordReset() {
+  alertNotImplemented();
```
Comment (avolkovi) [inline packages-exp/auth-exp/demo/src/index.js:787]: this is implemented now

File: packages-exp/auth-exp/demo/src/index.js [hunk 9]
```diff
@@ -0,0 +1,1905 @@
...
+function onGetProviderData() {
+  log('Providers data:');
+  log(activeUser()['providerData']);
+}
+
+/**
+ * Links a signed in user with an email and password account.
+ */
+function onLinkWithEmailAndPassword() {
+  alertNotImplemented();
```
Comment (avolkovi) [inline packages-exp/auth-exp/demo/src/index.js:825]: this is implemented now

File: packages-exp/auth-exp/demo/public/index.html [hunk 10]
```diff
@@ -0,0 +1,762 @@
...
+    <meta charset="UTF-8">
+    <title>Headless App</title>
+    <link rel="manifest" href="manifest.json">
+
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.12.0/jquery.min.js"></script>
+    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.6/js/bootstrap.min.js"
+            integrity="sha256-KXn5puMvxCw+dAYznun+drMdG1IFl3agK0p/pqT9KAo= sha512-2e8qq0ETcfWRI4HJBzQiA3UoyFk6tbNyG+qSaIBZLyW9Xf3sWZHN/lxe9fTh1U45DpPf07yj94KsUHHWe4Yk1A=="
+            crossorigin="anonymous"></script>
+    <script src="dist/bundle.js"></script>
+    <link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.6/css/bootstrap.min.css"
```
[the remark this answers is not in the document]
Comment (samhorlbeck) [reply packages-exp/auth-exp/demo/public/index.html:13]: Ack.

**Pull request merged**